ext/bootsnap/bootsnap.c

```diff
@@ -368,7 +368,8 @@ open_current_file(char * path, struct bs_cache_key * key, const char ** errno_pr
 
   fd = open(path, O_RDONLY);
   if (fd < 0) {
-    *errno_provenance = "bs_fetch:open_current_file:open";
+    // ensure error for reading the input file contains the original path
+    *errno_provenance = path;
```

Comment:
So `open_current_file` is only called from `bs_fetch`, and `path` here is `RSTRING_PTR` of the `path_v` `bs_fetch` received.

If we return `fd <0`, `bs_fetch` jumps to the `fail_errno` label, so we're not actually giving more information to the caller here, since it already have `path_v` on its stack.

If we wish to augment or change the error message, we can directly do it there.